src/net.c

```diff
@@ -2366,6 +2366,10 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
       response = NULL;
     } else {
       coap_log(LOG_WARNING, "cannot generate response\r\n");
+      if (response) {
```

Comment:
The delete functions in libcoap are all designed to be callable with a NULL argument, analogous to `free()`. Therefore, we can do without the extra check here.

Reply:
Code updated and pushed.